Dmitrii add ide docs

DmitriiKolesnichenko opened the pull request.

Commit:
doc: Add section "IDE"

Commit:
doc: Import page about ARC Project Templates from wiki

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/ARC-Project-Templates

Commit:
doc: Import page about building Linux uClibc applications from wiki

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Building-Linux-uClibc-applications

Commit:
doc: Import page about building user guide

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Building-User-Guide

Commit:
doc: Import page about creating a debug configuration

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Creating-a-Debug-Configuration

Commit:
doc: Import page about creating Eclipse plugins release zip file

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Creating-Eclipse-plugins-release-zip-file

Commit:
doc: Import page about debugging with custom GDB server

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Debugging-with-Custom-GDB-Server

Commit:
doc: Import page about debugging with nSIM

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Debugging-with-nSIM

Commit:
doc: Import page about debugging with OpellaXD

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Debugging-with-OpellaXD

Commit:
doc: Import page about debugging with OpenOCD

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Debugging-with-OpenOCD

Commit:
doc: Import page about debugging with running GDB server

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Debugging-with-Running-GDB-Server

Commit:
doc: Import page "Getting started with EM Starter Kit"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Getting-Started-with-EM-Starter-Kit

Commit:
doc: Import page "Getting Started with nSIM"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Getting-Started-with-nSIM

Commit:
doc: Import page "How to Use OpenOCD on Windows"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/How-to-Use-OpenOCD-on-Windows

Commit:
doc: Import page "Installation"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Installation

Commit:
doc: Import page "QfTest Test Suite User Manual"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/QfTest-Test-Suite-User-Manual

Commit:
doc: Import page "Using Debug Perspective"

Copied from:
https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/wiki/Using-Debug-Perspective

anthony-kolesov commented:
Can't build PDF from this branch, please check this.

First commit that fails to build is `doc: Import page "Installation"`

Commit:
doc: Increase limitations on the number of floats

Use package "morefloats" for increase limitations on the number of floats.
LaTeX by default can cope with maximum 18 floats. Add option for fixed image
location in document.

anthony-kolesov commented:
- [x] Please fix invalid location of text on IDE index page in PDF files.
- [x] PNGcrush pngs and resave JPGs so theat they'd use less space.

anthony-kolesov commented:
That has been merged.
